choreographer/src/main/java/eu/arrowhead/core/choreographer/service/ChoreographerExecutorService.java

```diff
@@ -150,21 +160,32 @@ public void unregisterExecutorSystem(final String name, final String origin) { /
     // assistant methods
 
 	//-------------------------------------------------------------------------------------------------
-	private void checkExecutorRequestDTO(final ChoreographerExecutorRequestDTO dto, final String origin, final HttpServletRequest servletRequest) {
+	private void checkAndNormalizeExecutorRequestDTO(final ChoreographerExecutorRequestDTO dto, final String origin, final HttpServletRequest servletRequest) {
 		logger.debug("checkExecutorRequestDTO started...");
 
 		// Check SystemRequestDTO only for nulls. (Verification will be done by ServiceRegistry)
 		if (dto == null) {
 			throw new BadPayloadException("Request is null.", HttpStatus.SC_BAD_REQUEST, origin);
 		}
 		
-		//TODO: what if system is null?
+		if (dto.getSystem() == null) {
+			throw new BadPayloadException("System is null.", HttpStatus.SC_BAD_REQUEST, origin);
+		}
 
 		if (Utilities.isEmpty(dto.getSystem().getSystemName())) {
 			throw new BadPayloadException("System name is empty.", HttpStatus.SC_BAD_REQUEST, origin);
 		}
 		
-		if (Utilities.isEmpty(dto.getSystem().getAddress())) {
+		if (!Utilities.isEmpty(dto.getSystem().getAddress())) {
+			try {
+				final String normalizedAddress = networkAddressPreProcessor.normalize(dto.getSystem().getAddress());
+				networkAddressVerifier.verify(normalizedAddress);
+				dto.getSystem().setAddress(normalizedAddress);
+			} catch (final InvalidParameterException ex) {
+				throw new BadPayloadException(ex.getMessage(), HttpStatus.SC_BAD_REQUEST, origin);
+			}			
+			
+		} else {
```

Comment:
This comment about using cnVerifier on service definition names (but I can't put to the line, because it not changed since last review). This is something the SR only does if the use_strict_service_definition_verifier property is set to true. So, you should ask this property from SR too, and use here.